src/core/src/OmniTileset.cpp

```diff
@@ -509,6 +528,57 @@ void OmniTileset::addImageryIon(const pxr::SdfPath& imageryPath) {
     _imageryPaths.push_back(imageryPath);
 }
 
+void OmniTileset::addImageryPolygon(const pxr::SdfPath& imageryPath) {
+    const OmniPolygonImagery imagery(imageryPath);
+
+    const auto uniqueName = "imagery_polygon_test"; // DEVEL
+
+    auto polygonImagery = UsdUtil::getCesiumPolygonImagery(imageryPath);
+    auto cartographicPolygonsRel = polygonImagery.GetCartographicPolygonBindingRel();
+    pxr::SdfPathVector cartographicPolygonTargets;
+    cartographicPolygonsRel.GetTargets(&cartographicPolygonTargets);
+    std::vector<CesiumGeospatial::CartographicPolygon> polygons;
+    for (const auto& cartographicPolygonTarget : cartographicPolygonTargets) {
+        auto cartographicPolygon = UsdUtil::getCesiumCartographicPolygon(cartographicPolygonTarget);
+        auto basisCurves = UsdUtil::getUsdBasisCurves(cartographicPolygonTarget);
+        auto pointsAttr = basisCurves.GetPointsAttr();
+        pxr::VtArray<pxr::GfVec3f> points;
+        pointsAttr.Get(&points);
+        auto anchor = UsdUtil::getCesiumGlobeAnchor(cartographicPolygonTarget);
+        auto origin = UsdUtil::getCartographicOriginForAnchor(anchor.GetPath());
+
+        std::vector<std::optional<CesiumGeospatial::Cartographic>> cartographicPositions;
+        auto transform = UsdUtil::computeUsdLocalToEcefTransformForPrim(origin.value(), cartographicPolygonTarget);
+
+        std::vector<glm::dvec2> polygon;
+        for (auto& curvePoint : points) {
+            auto homogenous = glm::dvec4(curvePoint[0], curvePoint[1], curvePoint[2], 1.0);
+            auto cartesianHomogenous = transform * homogenous;
+            auto cartesian = glm::dvec3(cartesianHomogenous.x, cartesianHomogenous.y, cartesianHomogenous.z);
+            std::optional<CesiumGeospatial::Cartographic> cartographicPosition =
+                CesiumGeospatial::Ellipsoid::WGS84.cartesianToCartographic(cartesian);
+
+            cartographicPositions.push_back(cartographicPosition);
+            polygon.emplace_back(glm::dvec2(cartographicPosition->longitude, cartographicPosition->latitude));
+        }
+        polygons.emplace_back(polygon);
+    }
```

Comment:
Potentially out of scope for this PR, and more relevant for my cleanup PR, I think a lot of this would belong in a `OmniCartographicPolygon` class.

Reply:
Agreed. Adding a new types of imagery/raster-overlay should require ideally no changes in `OmniTileset`